src/Framework/src/Microsoft.AspNetCore.App.Runtime.csproj

```diff
@@ -127,17 +130,21 @@ This package is an internal implementation of the .NET Core SDK and is not meant
 
     <Reference Include="microsoft.netcore.app.runtime.$(RuntimeIdentifier)" ExcludeAssets="All" PrivateAssets="All" />
 
-    <ProjectReference Condition=" '$(BuildIisNativeProjects)' == 'true' "
+    <ProjectReference Condition=" '$(BuildIisNativeProjects)' == 'true' AND $(BuildNative) "
         Include="$(RepoRoot)src\Servers\IIS\AspNetCoreModuleV2\InProcessRequestHandler\InProcessRequestHandler.vcxproj">
-      <SetPlatform>Platform=$(TargetArchitecture)</SetPlatform>
-      <SetPlatform Condition="'$(TargetArchitecture)' == 'x86'">Platform=Win32</SetPlatform>
+      <SetPlatform>Platform=$(NativePlatform)</SetPlatform>
       <!-- Custom attribute used to distinguish managed from native references. -->
       <IsNativeImage>true</IsNativeImage>
       <!-- A custom item group to be used in targets later.  -->
       <OutputItemType>_ResolvedNativeProjectReferencePaths</OutputItemType>
       <!-- C++ projects don't invoke GetTargetPath by default when building. -->
       <Targets>Build;GetTargetPath</Targets>
     </ProjectReference>
+
+    <_ResolvedNativeProjectReferencePaths Condition=" '$(BuildIisNativeProjects)' == 'true' AND !$(BuildNative) "
```

Comment:
Unlike other managed projects with references to native projects, this one didn't have a fallback to find the right files and pack them up.